linc_convert/utils/io/zarr/drivers/zarr_python.py

```diff
@@ -0,0 +1,413 @@
+
+
+class ZarrPythonArray(ZarrArray):
+    """Zarr Array implementation using the zarr-python library."""
+
+    def __init__(self, array: zarr.Array) -> None:
+        """
+        Initialize the ZarrPythonArray with a zarr.Array.
+
+        Parameters
+        ----------
+        array : zarr.Array
+            Underlying Zarr array.
+        """
+        super().__init__(str(array.store_path))
+        self._array = array
+
+    @property
+    def ndim(self) -> int:
+        """Number of dimensions of the array."""
+        return self._array.ndim
+
+    @property
+    def shape(self) -> Tuple[int, ...]:
+        """Shape of the array."""
+        return self._array.shape
+
+    @property
+    def dtype(self) -> np.dtype:
+        """Data type of the array."""
+        return self._array.dtype
+
+    @property
+    def chunks(self) -> Tuple[int, ...]:
+        """Chunk shape for the array."""
+        return self._array.chunks
+
+    @property
+    def shards(self) -> Optional[Tuple[int, ...]]:
+        """Shard shape, if supported; otherwise None."""
+        return getattr(self._array, "shards", None)
+
+    @property
+    def attrs(self) -> Mapping[str, Any]:
+        """Access metadata/attributes for this node."""
+        return self._array.attrs
+
+    @property
+    def zarr_version(self) -> int:
+        """Get the Zarr format version."""
+        return self._array.metadata.zarr_format
+
+    def __getitem__(self, key: str) -> ArrayLike:
+        """Read data from the array."""
+        return self._array[key]
+
+    def __setitem__(self, key: str, value: ArrayLike | Number) -> None:
+        """Write data to the array."""
+        self._array[key] = value
+
+    def __getattr__(self, name: str) -> Any:  # noqa: ANN401
+        """Delegate any unknown attributes to the underlying array."""
+        if name == "_array":
+            return self._array
+        if hasattr(self._array, name):
+            return getattr(self._array, name)
+        raise AttributeError(
+                f"'{self.__class__.__name__}' object has no attribute '{name}'"
+                )
+
+
+class ZarrPythonGroup(ZarrGroup):
+    """Zarr Group implementation using the zarr-python library."""
+
+    def __init__(self, zarr_group: zarr.Group) -> None:
+        """
+        Initialize the ZarrPythonGroup with a zarr.Group.
+
+        Parameters
+        ----------
+        zarr_group : zarr.Group
+            Underlying Zarr Python group.
+        """
+        super().__init__(str(zarr_group.store_path))
+
+        self._zgroup = zarr_group
+
+    @classmethod
+    def from_config(cls, zarr_config: ZarrConfig) -> "ZarrPythonGroup":
+        """Create a Zarr group from a configuration object."""
+        store = zarr.storage.LocalStore(zarr_config.out)
+        return cls(
+                zarr.group(
+                        store=store,
+                        # TODO: figure out overwrite
+                        # overwrite=overwrite,
```

Comment:
We should figure this out before merging. Can we use classic opening modes like "r", "w", "a", "r+" in the config, and set overwrite accordingly? ("w" > overwrite=True, others > overwrite=False)